packages/validator/src/services/validatorStore.ts

```diff
@@ -168,14 +168,46 @@ export class ValidatorStore {
   }
 
   getFeeRecipient(pubkeyHex: PubkeyHex): Eth1Address {
-    return this.validators.get(pubkeyHex)?.feeRecipient ?? this.defaultProposerConfig.feeRecipient;
+    if (this.validators.has(pubkeyHex)) {
+      return this.validators.get(pubkeyHex)?.feeRecipient ?? this.defaultProposerConfig.feeRecipient;
+    } else {
+      throw Error(`Validator pubkey ${pubkeyHex} not known`);
+    }
   }
 
   getFeeRecipientByIndex(index: ValidatorIndex): Eth1Address {
     const pubkey = this.indicesService.index2pubkey.get(index);
     return pubkey ? this.getFeeRecipient(pubkey) : this.defaultProposerConfig.feeRecipient;
   }
 
+  setFeeRecipient(pubkeyHex: PubkeyHex, feeRecipient: Eth1Address): void {
+    if (this.validators.has(pubkeyHex)) {
+      const validatorData = this.validators.get(pubkeyHex);
+      if (validatorData !== undefined) {
+        //const newValidatorData = {...validatorData, feeRecipient}
+        this.validators.set(pubkeyHex, {...validatorData, feeRecipient});
+      } else {
+        throw Error(`ValidatorData for pubkey ${pubkeyHex} not set `);
+      }
+    } else {
+      throw Error(`Validator pubkey ${pubkeyHex} not known`);
+    }
```

Comment:
Don't use has, then get. Call get then check for undefined. Please apply this comment to the entire PR